sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SparkPlanGraph.scala

```diff
@@ -101,12 +104,15 @@ private[sql] object SparkPlanGraph {
         } else {
           subgraph.nodes += node
         }
+        if (name == "ShuffleExchange" || name == "BroadcastExchange") {
```

Comment:
I'm glad to revert this change.

Btw, it's hard to protect future change anyway (because of unknown).